kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -5932,7 +5936,14 @@ public void commitResource(ContentResourceEdit edit, int priority) throws OverQu
                 //This might not want to be set as it would advise the detector
                 metadata.set(Metadata.RESOURCE_NAME_KEY, edit.getId());
                 metadata.set(Metadata.CONTENT_TYPE, currentContentType);
-                String newmatch = DETECTOR.detect(TikaInputStream.get(buff), metadata).toString();
+                String newmatch = "";
+                //If we are ignoring the content for this extension, don't give it any data
+                if (m_ignoreContent != null && Arrays.asList(m_ignoreContent).contains(FilenameUtils.getExtension(edit.getId()))) {
```

Comment:
As you're only ever treating it as an `List` when searching through it you could just make the member a `List`.